src/test/java/com/redhat/darcy/webdriver/WebDriverCookieManagerTest.java

```diff
@@ -0,0 +1,172 @@
+    private WebDriver.Options mockOptions = mock(WebDriver.Options.class);
+    private WebDriverCookieManager cookieManager = new WebDriverCookieManager(mockDriver);
+
+    @Before
+    public void stubDriver() {
+        when(mockDriver.manage())
+                .thenReturn(mockOptions);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameAndValue() {
+        Cookie cookie = new Cookie("chocolate", "chip");
+        org.openqa.selenium.Cookie seleniumCookie =
+                new org.openqa.selenium.Cookie("chocolate", "chip");
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValueAndPath() {
+        Cookie cookie = new Cookie("chocolate", "chip", null);
+        org.openqa.selenium.Cookie seleniumCookie =
+                new org.openqa.selenium.Cookie("chocolate", "chip", null);
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValuePathAndExpiry() {
+        Instant instantNow = Instant.now();
+        Cookie cookie = new Cookie("chocolate", "chip", "home", LocalDateTime.ofInstant(instantNow,
+                ZoneId.systemDefault()));
+        org.openqa.selenium.Cookie seleniumCookie =
+                new org.openqa.selenium.Cookie("chocolate", "chip", "home", Date.from(Instant.now()));
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValuePathAndNullExpiry() {
+        Cookie cookie = new Cookie("chocolate", "chip", "home", null);
+        org.openqa.selenium.Cookie seleniumCookie =
+                new org.openqa.selenium.Cookie("chocolate", "chip", "home", null);
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldProperlyAddACookieWithANameValueDomainPathAndExpiry() {
+        Instant instantNow = Instant.now();
+        Cookie cookie = new Cookie("chocolate", "chip", null, "", LocalDateTime.ofInstant(instantNow,
+                ZoneId.systemDefault()));
+        org.openqa.selenium.Cookie seleniumCookie =
+                new org.openqa.selenium.Cookie("chocolate", "chip", null, "", Date.from(instantNow));
+
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(seleniumCookie);
+    }
+
+    @Test
+    public void shouldGetCookie() {
+        Instant instantNow = Instant.now();
+        Cookie cookie = new Cookie("chocolate", "chip", null, "", LocalDateTime.ofInstant(instantNow,
+                ZoneId.systemDefault()));
+        org.openqa.selenium.Cookie seleniumCookie = new org.openqa.selenium.Cookie("chocolate",
```

Comment:
Please pass actual values for domain and path and also include the boolean flags. We want to test everything.